spec/rubocop/cop/rspec/leaky_constant_declaration_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::LeakyConstantDeclaration do
+  subject(:cop) { described_class.new }
+
+  describe 'constant assignment' do
+    it 'flags inside an example group' do
+      bad_code = <<-RUBY
+        describe SomeClass do
+          CONSTANT = 'is not allowed'.freeze
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'flags inside shared example group' do
+      bad_code = <<-RUBY
+        RSpec.shared_examples 'shared example' do
+          CONSTANT = 'is not allowed'.freeze
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'flags inside an example' do
+      bad_code = <<-RUBY
+        describe SomeClass do
+          specify do
+            CONSTANT = 'is not allowed'.freeze
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+          end
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'ignores outside of example/shared group' do
+      fair_code = <<-RUBY
+        factory :some_class do
+          CONSTANT = 'is not allowed'.freeze
```

Comment:
```suggestion
          CONSTANT = 'is allowed'.freeze
```

Reply:
I'll change this to something more self-explanatory, e.g. `'I will be accessible as ::CONSTANT'`, or neutral, e.g. "I'm a constant", what do you think?

Reply:
Yes, that’s probably better. 👍